scripts/dump-users.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const program = require('commander')
+
+program
+  .option('-e, --emails <emails>', 'Email addresses to dump, comma separated')
+  .option('-u, --uids <uids>', 'User IDs to dump, comma separated')
+  .option('-p, --pretty', 'Display output in a pretty fashion')
+  .parse(process.argv)
+
+const emails = parseEmails(program.emails)
+const uids = parseUids(program.uids)
+
+if (! emails.length && ! uids.length) {
+  console.error('One of `emails` or `uids` must be specified')
+  process.exit(1)
+} else if (emails.length && uids.length) {
+  console.error('Only one of `emails` or `uids` can be specified, not both')
+  process.exit(1)
+}
+
+const iterateOver = emails.length ? emails : uids
+const dbFunc = emails.length ? 'emailRecord': 'account'
+
+const config = require('../config').getProperties()
+const log = {
+  error: (msg) => {},
+  info: (msg) => {},
+  trace: (msg) => {},
+}
+
+const Token = require('../lib/tokens')(log, config)
+const UnblockCode = require('../lib/crypto/random').base32(config.signinUnblock.codeLength)
+const P = require('../lib/promise')
+
+const DB = require('../lib/db')(
+  config,
+  log,
+  Token,
+  UnblockCode
+)
+
+let db;
+
+DB.connect(config[config.db.backend])
+  .then(_db => {
+    db = _db;
+    return P.mapSeries(iterateOver, (item) => db[dbFunc](item))
+  }).then((records) => {
+    if (program.pretty) {
+      console.log(JSON.stringify(records, null, 2))
+    } else {
+      console.log(JSON.stringify(records))
+    }
+
+    return db.close()
+  })
+  .then(() => {
+    process.exit(0)
+  })
+
+
+function parseEmails(emails = '') {
+  if (! emails.length) {
```

Comment:
Should we impose some limit on how many emails/uid can be dumped at once?

Reply:
It would be a somewhat flimsy roadblock, a malicious user with access to the server would just run the script again or dump the DB by running a query.

Reply:
I was thinking it would be more for us, if we are trying to dump 1000 users and slow down database. Although, I guess if it is done on a read replica then it wouldn't really matter 🤷🏽‍♂️